pyo3-macros-backend/src/pyclass.rs

```diff
@@ -230,34 +230,52 @@ pub fn build_py_class(
             For an explanation, see https://pyo3.rs/latest/class.html#no-generic-parameters"
     );
 
-    let mut field_options: Vec<(&syn::Field, FieldPyO3Options)> = match &mut class.fields {
-        syn::Fields::Named(fields) => fields
-            .named
-            .iter_mut()
-            .map(|field| {
-                FieldPyO3Options::take_pyo3_options(&mut field.attrs)
-                    .map(move |options| (&*field, options))
-            })
-            .collect::<Result<_>>()?,
-        syn::Fields::Unnamed(fields) => fields
-            .unnamed
-            .iter_mut()
-            .map(|field| {
-                FieldPyO3Options::take_pyo3_options(&mut field.attrs)
-                    .map(move |options| (&*field, options))
-            })
-            .collect::<Result<_>>()?,
-        syn::Fields::Unit => {
-            if let Some(attr) = args.options.set_all {
-                return Err(syn::Error::new_spanned(attr, UNIT_SET));
-            };
-            if let Some(attr) = args.options.get_all {
-                return Err(syn::Error::new_spanned(attr, UNIT_GET));
-            };
-            // No fields for unit struct
-            Vec::new()
-        }
-    };
+    let mut field_options_res: Vec<Result<(&syn::Field, FieldPyO3Options)>> =
+        match &mut class.fields {
+            syn::Fields::Named(fields) => fields
+                .named
+                .iter_mut()
+                .map(|field| {
+                    FieldPyO3Options::take_pyo3_options(&mut field.attrs)
+                        .map(move |options| (&*field, options))
+                })
+                .collect::<Vec<_>>(),
+            syn::Fields::Unnamed(fields) => fields
+                .unnamed
+                .iter_mut()
+                .map(|field| {
+                    FieldPyO3Options::take_pyo3_options(&mut field.attrs)
+                        .map(move |options| (&*field, options))
+                })
+                .collect::<Vec<_>>(),
+            syn::Fields::Unit => {
+                if let Some(attr) = args.options.set_all {
+                    return Err(syn::Error::new_spanned(attr, UNIT_SET));
+                };
+                if let Some(attr) = args.options.get_all {
+                    return Err(syn::Error::new_spanned(attr, UNIT_GET));
+                };
+                // No fields for unit struct
+                Vec::new()
+            }
+        };
+
+    // handle error here
+
+    let mut all_error = ErrorCombiner(None);
+
+    let mut field_options: Vec<(&syn::Field, FieldPyO3Options)> = field_options_res
+        .drain(..)
+        .filter_map(|result| match result {
+            Err(err) => {
+                all_error.combine(err);
+                None
+            }
+            Ok(options) => Some(options),
+        })
+        .collect::<Vec<_>>();
```

Comment:
I think we should now be able to simplify here and go back to something more like the original implementation and drop the need for the intermediate `field_options_res`.

I think the idea would be to set up `all_errors` before the `match &mut class.fields`. And then instead of collecting into vec-of-results I think it would be possible to swap the `.map` for `.filter_map`.

Something like

```rust
                .filter_map(|field| {
                    match FieldPyO3Options::take_pyo3_options(&mut field.attrs) {
                        Ok(options) => Some((&*field, options)),
                        Err(e) => {
                            all_errors.combine(e);
                            None
                        }
                    }                    
                })
```